src/test/java/com/github/kongchen/swagger/docgen/reader/JaxrsReaderTest.java

```diff
@@ -102,6 +104,21 @@ private void assertSwaggerResponseContents(Tag expectedTag, Swagger result) {
         assertFalse(path.getOperations().isEmpty(), "Should be a get operation");
     }
 
+    @Test
+    public void discoverSubResource() {
+        Swagger result = reader.read(SomeResource.class);
+        assertSwaggerPath(result.getPath("/resource/explicit/name").getGet(), result, "/resource/implicit/name");
+    }
+
+    private void assertSwaggerPath(Operation expectedOperation, Swagger result, String expectedPath) {
+        assertNotNull(result, "No Swagger object created");
+        assertFalse(result.getPaths().isEmpty(), "Should contain operation paths");
+        assertTrue(result.getPaths().containsKey(expectedPath), "Expected path missing");
+        io.swagger.models.Path path = result.getPaths().get(expectedPath);
```

Comment:
If I'm reading this correctly, it appears you are asserting on the expected, not the actual. Was this intended?